docs/source/features.md

```diff
@@ -181,24 +181,80 @@ negate'' = \case
     False -> True
 ```
 
-## Partially-Applied Case Pattern Match
+## Flow Control
 
+Agda2HS provides native support for the Haskell `if_then_else_` and `case_of_` constructs.
+
+Agda:
 ```agda
-len : List a → Int
-len xs = case xs of length
-{-# COMPILE AGDA2HS len #-}
+ifThenElse : Int → String
+ifThenElse n = if n >= 10 then "big" else "small"
+{-# COMPILE AGDA2HS ifThenElse #-}
 
-applyToFalse : (Bool → a) → a
-applyToFalse = case False of_
-{-# COMPILE AGDA2HS applyToFalse #-}
+mhead : List a → Maybe a
+mhead xs = case xs of λ where
+  []      → Nothing
+  (x ∷ _) → Just x
+{-# COMPILE AGDA2HS mhead #-}
 ```
 
-```hs
-len :: List a -> Int 
-len xs = length xs
+Haskell:
+```haskell
+ifThenElse :: Int -> String
+ifThenElse n = if n >= 10 then "big" else "small"
+
+mhead :: [a] -> Maybe a
+mhead xs
+  = case xs of
+        [] -> Nothing
+        x : _ -> Just x
+```
+
+> **It is NOT possible to partially apply these two constructs.**
+> This means that you must explicitly write `λ x → if x then 2 else 3` instead of `if_then 2 else 3`. (This copies the impossibility of the second implementation in Haskell.)
```

Comment:
One thing we could do is to compile a partially applied `case_of \where ...` to a `\case` in Haskell (this would also require enabling the LambdaCase extension). But this doesn't necessarily have to be part of this PR.